src/modules/river/tags.cpp

```diff
@@ -189,11 +189,18 @@ bool Tags::handle_button_press(GdkEventButton *event_button, uint32_t tag) {
 }
 
 void Tags::handle_focused_tags(uint32_t tags) {
+  auto hide_vacant = config_["hide-vacant"].asBool();
   for (size_t i = 0; i < buttons_.size(); ++i) {
     if ((1 << i) & tags) {
+      if (hide_vacant) {
+        buttons_[i].set_visible(true);
+      }
       buttons_[i].get_style_context()->add_class("focused");
     } else {
       buttons_[i].get_style_context()->remove_class("focused");
+      if (hide_vacant && !buttons_[i].get_style_context()->has_class("occupied")) {
```

Comment:
This is a bit hacky but I'm not sure of a better solution. 

Without this extra condition, tags will remain permanently visible as long as they have been focused.